### ISSUE TITLE
How to close popover on button click

### ISSUE THREAD

**Issue by BlagojeV93** [doomsower/react-native-modal-popover#22]
I have a CANCEL button on popover, and I'm trying to close it when its open on CANCEL click. Any tips how. Tried combinig state and visible prop, but that doesn't seem to work

**Comment by baba43** [comment]
I have a similar problem.

**Comment by doomsower** [comment]
Are you using `PopoverTouchable`?

**Comment by baba43** [comment]
Yes. I have some kind of menu within the Popup (nested `TouchableWithoutFeedback`).

The popup is closing fine when clicking somewhere else, so that is no problem.

**Comment by doomsower** [comment]
`PopoverTouchable` is for simple self-closing popovers. For behavior you want you should use `Popover` without `PopoverTouchable` and implement it manually. See `PopoverTouchable` for hints on how to do this.

So I see this is popular request, and this is not the first issue about it, so I'll think about a adding this feature

**Comment by baba43** [comment]
Thank you. For now I have figured out this solution:

```javascript
function MyComponent(props) {
  let modal;
  return (
    <PopoverTouchable
      ref={m => {
        modal = m;
      }}
    >
      // ...
      <Popover>
        <TouchableWithoutFeedback
          onPress={() => {
            modal.onClosePopover();
          }}
        />
      </Popover>
    </PopoverTouchable>
  );
}
```

**Comment by BlagojeV93** [comment]
Thanks for this, it works!

**Comment by doomsower** [comment]
I just published version `0.0.7` of this package. It introduces `PopoverController` which is intended to replace `PopoverTouchable` (which is still there, if you worry). It offers you more control over popover behavior. However, it uses render prop pattern, which some people don't like. Please check it out.

I also updated readme with an example of pure `Popover` usage.
